### ISSUE TITLE
Not matching line numbers with source java code

### ISSUE THREAD

**Issue by rameshchandra85** [java-decompiler/jd-gui#247]
Wherever decompiler shows the line number , it does show the correct line number with source. However overall line numbers doesn't match with source code. Is it possible to fix this ?

**Comment by emmanue1** [comment]
Could you attach a CLASS file ?

**Comment by rameshchandra85** [comment]
[UIModeler.zip](https://github.com/java-decompiler/jd-gui/files/3315711/UIModeler.zip)

**Comment by rameshchandra85** [comment]
Attached the class file inside the zip , if you open in jd-gui and scroll towards the end it shows the line number as 545 which is the actual line number. But if you save as source it has 322 lines

**Comment by emmanue1** [comment]
Hi, I drag and drop your file onto JD-GUI 1.6.1. I don't see any probleme on misalignment : 
![image](https://user-images.githubusercontent.com/11559196/60387706-a29e2580-9aa7-11e9-9e38-5e6c38971c78.png)
If a save the source, I get that:
![image](https://user-images.githubusercontent.com/11559196/60387747-4982c180-9aa8-11e9-8e26-faaa9c5eb4ba.png)
All looks OK, no ?

**Comment by rameshchandra85** [comment]
Yeah , I was using 1.4 version and that was not matching the lines when I export. I have tried with 1.6 version , now the line numbers do match , however I see there are lot of spacing added in between within the java source code (I've checked multiple class files and everywhere its adding lot of space gap in between the java source code). I'm sure the original source code won't have that space , is it just to match the line number ? . And also as you've mentioned switch case doesn't show up correctly.

**Comment by rameshchandra85** [comment]
[UIModeler_Src.zip](https://github.com/java-decompiler/jd-gui/files/3347381/UIModeler_Src.zip)

**Comment by emmanue1** [comment]
I agree with you: empty lines may seem strange at first.

If your goal is just to read the decompiled source code, you can uncheck the checkbox "Realign line numbers" in Preferences panel.  In this cas, no empty line is added. "Realign line numbers" is fine if your goal is to debug a JAR file or study a stack trace.

In most cases, the original source code has no empty lines but comments instead. You can check it with this experience :
1) Uncheck option "Search source code on maven.org for",
2) Drag and drop "log4j-core-2.6.2.jar" in JD-GUI 1.6.1 for example,
3) Display "org.apache.logging.log4j.core.Logger",
4) Now, recheck case "Search source code on maven.org for",
5) Close and reopen "org.apache.logging.log4j.core.Logger" if the original source code does not appear.
